service/history/historyEngine.go

```diff
@@ -756,10 +756,11 @@ func (e *historyEngineImpl) DescribeWorkflowExecution(
 				WorkflowId: common.StringPtr(executionInfo.WorkflowID),
 				RunId:      common.StringPtr(executionInfo.RunID),
 			},
-			Type:            &workflow.WorkflowType{Name: common.StringPtr(executionInfo.WorkflowTypeName)},
-			StartTime:       common.Int64Ptr(executionInfo.StartTimestamp.UnixNano()),
-			HistoryLength:   common.Int64Ptr(msBuilder.GetNextEventID() - common.FirstEventID),
-			AutoResetPoints: executionInfo.AutoResetPoints,
+			Type:             &workflow.WorkflowType{Name: common.StringPtr(executionInfo.WorkflowTypeName)},
+			StartTime:        common.Int64Ptr(executionInfo.StartTimestamp.UnixNano()),
+			HistoryLength:    common.Int64Ptr(msBuilder.GetNextEventID() - common.FirstEventID),
+			AutoResetPoints:  executionInfo.AutoResetPoints,
+			SearchAttributes: &workflow.SearchAttributes{IndexedFields: executionInfo.SearchAttributes},
```

Comment:
Could *executionInfo.SearchAttributes* be null?

Reply:
Yes it could, but that's fine.

Reply:
Then desc result will be weird: SearchAttributes will be not nil but it's content is nil. In this case usually we would just make SearchAttributes is nil, so that when reading this field, only need to check if searchAttr is nil or not